### PR TITLE
Add option to edit and add new groups

### LINKED ISSUE #91: GroupBar : Add option to edit groups and add new group.
### Describe the feature

This feature aims to provide the option to edit existing groups and create new groups through the web frontend.
This can only be perfom in the monitors page.

### Use Case

This feature is needed so that an administrator can edit groups of monitors in the system, and it as to bee simple task to do.

### Proposed Solution

The solution is in the GroupBar of monitors page having a button to enter in edit mode in this mode every text that can be edited is an input box so the user can edit. And appears a button to create a new Group. 

### Other Information

_No response_

### PR DESCRIPTION
**Pull request by fungame2270**
### Issue

Closes #91 

### Reason for this change
this change makes that we can change group name and description in frontend.

### Description of changes

In this pr the groupController was fixed the update didn't update description.
in the frontend all logic needed for updating and creating group was added.
Created service in frontend to update and create monitor

### Description of how you validated changes

the changes were validated by testing in frontend

### Checklist
- [x] I have tested my changes locally.
- [ ] I have updated the documentation to reflect my changes.
- [ ] I have added/updated unit tests (if applicable).
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://ua-smart-signage-platform.github.io/Documentation-Website/guidelines/)

### CODE REVIEW

File: frontend/src/components/GroupBar/index.jsx [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import { useEffect, useState } from "react";
-import { MdSettings, MdCreate } from "react-icons/md";
+import { MdSettings, MdCreate,MdAdd,MdCheck } from "react-icons/md";
```
Comment (pdPinho) [inline frontend/src/components/GroupBar/index.jsx]: Either keep spaces between imports or keep everything together

File: frontend/src/components/GroupBar/index.jsx [hunk 2]
```diff
@@ -13,26 +15,80 @@ function GroupBar( {id, changeId, page} ) {
...
     }
 
+    const changeEditGroupName = (name) =>{
+        setEditGroup({
+            id:editGroup.id,
+            name: name,
+            description: editGroup.description,
+        })
+    }
+    const cahngeEditGroupDescription = (description) =>{
```
Comment (pdPinho) [inline frontend/src/components/GroupBar/index.jsx]: Change `cahngeEditGroupDescription` to `editGroupDescription`

File: frontend/src/components/GroupBar/index.jsx [hunk 3]
```diff
@@ -13,26 +15,80 @@ function GroupBar( {id, changeId, page} ) {
...
     }
 
-    const changeInformation = () => {
-        
+    const setupCreateGroup = () =>{
+        if(!groups.some((element)=> element.id == -1))
+            setGroups([...groups,{id:-1,name:"",description:""}])
     }
 
+    const changeEditGroupName = (name) =>{
```
Comment (pdPinho) [inline frontend/src/components/GroupBar/index.jsx]: Change `changeEditGroupName` to `editGroupName`

File: frontend/src/components/GroupBar/index.jsx [hunk 4]
```diff
@@ -13,26 +15,80 @@ function GroupBar( {id, changeId, page} ) {
...
+    }
+    const cahngeEditGroupDescription = (description) =>{
+        setEditGroup({
+            id:editGroup.id,
+            name: editGroup.name,
+            description: description,
+        })
+    }
+    
+    const handleUpdateCreate = () =>{
```
Comment (pdPinho) [inline frontend/src/components/GroupBar/index.jsx:45]: Change `handleUpdateCreate` to `handleUpdateGroup` or `handleUpdateGroupInformation`

File: frontend/src/components/GroupBar/index.jsx [hunk 5]
```diff
@@ -13,26 +15,80 @@ function GroupBar( {id, changeId, page} ) {
...
+        if (editGroup.id < 0){
+            monitorsGroupService.createGroup(editGroup).then((response) =>{
+                monitorsGroupService.getGroups().then((groupsData) => {
+                    setGroups(groupsData.data);
+                })
+            })
+        }else{
```
Comment (pdPinho) [inline frontend/src/components/GroupBar/index.jsx]: Either keep every `if... else` condition across every file with this aspect or put `else` below the bracket

**Comment by pdPinho** [comment]
Don't forget to address the reliability Sonarcloud issues.

File: frontend/src/components/GroupBar/index.jsx (hunk 3 above)
Comment (fungame2270) [reply frontend/src/components/GroupBar/index.jsx]: Ele da Change do nome na variável editgroup

**Review by pdPinho — APPROVED** [review]
Looks fine to me :+1:

**Pull request merged**